android/library/src/main/java/com/localytics/library/MediaTracker.java

```diff
@@ -69,22 +71,45 @@ public void complete() {
 
     private void mergeOrCreateRange(int watchStart, int watchEnd) {
         if (watchStart < 0) {
-            throw new IllegalStateException("The video can't be tagged as stopped before it is tagged as started.");
+            Log.i("LocalyticsMediaTracker", "Media Tracker dropping stop datapoint.  A stop was " +
+                    "called before a start, this can occur when a start was never called or stop " +
+                    "was called multiple times in a row.");
+            return;
         }
-        if (watchEnd > videoDurationMS || watchStart > videoDurationMS) {
-            throw new IllegalArgumentException("The time input is greater than the duration of the media.");
+        if (watchEnd > videoDurationMS) {
+            Log.i("LocalyticsMediaTracker", "Media Tracker dropping stop datapoint.  A stop was " +
+                    "called with a time that is larger than the media duration.");
+            return;
         }
         boolean overlap = false;
         int i = 0;
         while (!overlap && i < rangesWatched.size()) {
-            overlap = rangesWatched.get(i++).mergeIfOverlapping(watchStart, watchEnd);
+            Range range = rangesWatched.get(i++);
+            overlap = range.mergeIfOverlapping(watchStart, watchEnd);
+            if (overlap) {
+                ensureNewRangehasNoOverlap(range);
+            }
         }
         if (!overlap) {
             rangesWatched.add(new Range(watchStart, watchEnd));
         }
         startedTime = -1;
     }
 
+    private void ensureNewRangehasNoOverlap(Range input)
```

Comment:
I'm trying to grok this method.  The name of it is throwing me off.   What does it do?

- I'm guessing here, but would a better name be `removeOverlappingRanges` ?
- can we make the inputs more explicit and pass in `rangesWatched`? 
- if we pass in all inputs, then unit test this method should be trivial
- unit test would have the added benefit of describing the expected behavior.